### ISSUE TITLE
Support .Net Core

### ISSUE THREAD

**Issue by haseeb1431** [enyim/EnyimMemcached#155]
Any plans/Road map to support the .Net core in near future since it has released officially (#146)?

**Comment by cnblogs-dudu** [comment]
We migrated it to .NET Core. See https://www.nuget.org/packages/EnyimMemcachedCore

**Comment by derFunk** [comment]
@enyim What's your plan with official .NET Core support? On the roadmap?

**Comment by enyim** [comment]
It's on the roadmap, but i'm a bit busy with life, so I cannot say exact date. Probably in a couple of weeks I can start working on it. 

And I'm always open to pull requests. :)

My biggest problem is that it requires significant splitting of functionality and I have to change a lot of things (Or backport code from EnyimMemcached2.)

**Comment by derFunk** [comment]
@cnblogs-dudu would you create a PR from your branch?

**Comment by derFunk** [comment]
@enyim There's another big project (we're also using) which has a dependency to your lib: https://github.com/awslabs/elasticache-cluster-config-net - they're currently blocked to port to .NET Core because of that ;) (Although they could use a reference to @cnblogs-dudu fork already).

**Comment by enyim** [comment]
I'm open to all help, but I think the biggest improvement would come from MS ditching the dotnet.exe pipeline and switching back to msbuild :)

I'll see what I can do.

**Comment by enyim** [comment]
Please see the `dotnetcore` branch

- You have to use the `Enyim.Caching.Standard.sln` to build the code
- It produces a .NET 4.5 and a Netstandard 1.6 compatible output. Eventually this project will only produce the 1.6 version, but it's easier to test this way
- There is no BinaryFormatter in netstandard, so you will have to implement your own Transcoder
- There is no XML configuration API in netstandard (nor any usable structured config source), so you have to configure everything from code (most people use DI anyway). The most important impact here is that the default constructor of the MemcachedClient is gone. It may come back in some form or other.
- Logging adapters are not ported yet
- No DefaultPerformanceMonitor either
- No nuget packages yet

There are some minor changes in the code, but I tried to keep backward compatibiliity wherever possible.

Please test the code, see if it works, if anything you have used previously is missing, etc. Pull requests are welcome.

Overall, I'm not planning any big investments into .NET core/standard/whatever MS decides to call it next month until the APIs and the tooling finalizes, right now this is still a big mess.

**Comment by enyim** [comment]
Update

Vs 2017 RC is still crap, event a simple netstandard1.5+net45 project throws hundreds of warnings (and sometimes crashes VS)

good news: 

- we'll eventually have .NET Standard packages
- targeting Standard 1.5, normal .NET 4.5.0

bad news

- probably only after VS 2017 RTM

**Comment by MichaCo** [comment]
@enyim I think the tooling is now much more mature and netstandard2.0 also brings back binary serialization AND `System.Configuration`!
Performance monitors are currently in the works I think (preview package available).
Not sure about logging.
You'd still have to multi target for .NET 4.5 if .NET 4.6.1 as minimum target is too aggressive.

**Comment by luferogo** [comment]
> 
> 
> We migrated it to .NET Core. See https://www.nuget.org/packages/EnyimMemcachedCore

Hey @cnblogs-dudu , are u still maintaining that fork? I'd like to give it a try but just wanted to know if you guys would be available to attend any issue.  I appreciate your collaboration very much!

**Comment by cnblogs-dudu** [comment]
@luferogo Yes. We are using it in our production environment. We are maintaining it.
